exec/java-exec/src/main/resources/drill-module.conf

```diff
@@ -427,8 +427,8 @@ drill.exec.options: {
     exec.enable_union_type: false,
     exec.errors.verbose: false,
     exec.hashagg.mem_limit: 0,
-    exec.hashagg.min_batches_per_partition: 2,
-    exec.hashagg.num_partitions: 32,
+    exec.hashagg.min_batches_per_partition: 1,
```

Comment:
This option was meant to create a "slack". **1** is the lowest value - requiring only 1 batch per each partition, i.e., no slack; so that requires the memory computations to be more precise now !!

Reply:
@Ben-Zvi This setting controls the minimum number of batches kept in memory per partition. Making this larger will cause us to consume more memory. Making it smaller makes us consume less memory. Also in general the purpose of this PR was to make the memory calculations more precise and deterministic and it passes all regression tests.